File: CarritoBlueJ/CarritodeCompras/DetalleCarrito.java

public class DetalleCarrito
{
    private Producto producto;
    private int cantidad;

    public DetalleCarrito(Producto producto, int cantidad)
    {
        this.producto = producto;
        this.cantidad = cantidad;
    }

    public Producto getProducto()
    {
        return producto;
    }

    public int getCantidad()
    {
        return cantidad;
    }

    public void setCantidad(int cantidad)
    {
        this.cantidad = cantidad;
    }

    public double calcularSubtotal()
    {
        return producto.getPrecio() * cantidad;
    }

    public String toString()
    {
        return producto.getNombre() + " x" + cantidad + " = Q" + calcularSubtotal();
    }
}

File: CarritodeCompras/logica/DetalleCarrito.cs
using System;

namespace CarritodeCompras.logica
{
    public class DetalleCarrito
    {
        private Producto producto;
        private int cantidad;

        public DetalleCarrito(Producto producto, int cantidad)
        {
            this.producto = producto;
            this.cantidad = cantidad;
        }

        public Producto Producto { get { return producto; } }
        public int Cantidad { get { return cantidad; } }

        public double CalcularSubtotal()
        {
            return producto.Precio * cantidad;
        }

        public override string ToString()
        {
            return $"{producto.Nombre} x{cantidad} = Q{CalcularSubtotal():F2}";
        }
    }
}
